content/en/docs/tasks/traffic-management/egress/egress-kubernetes-services/index.md

```diff
@@ -0,0 +1,268 @@
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    kind: Endpoints
+    apiVersion: v1
+    metadata:
+      name: my-wikipedia
+    subsets:
+      - addresses:
+          - ip: 91.198.174.192
+          - ip: 198.35.26.96
+        ports:
+          - port: 443
+            name: tls
+    EOF
+    {{< /text >}}
+
+1.  Observe your service. Note that it has a cluster IP which you can use to access `wikipedia.org`.
+
+    {{< text bash >}}
+    $ kubectl get svc my-wikipedia
+    NAME           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+    my-wikipedia   ClusterIP   172.21.156.230   <none>        443/TCP   21h
+    {{< /text >}}
+
+1.  Send HTTPS requests to `wikipedia.org` by your Kubernetes service's cluster IP from the source pod without Istio
+    sidecar.
+    Use the `--resolve` option of `curl` to access `wikipedia.org` by the cluster IP:
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD_WITHOUT_ISTIO -n without-istio -c sleep -- curl -s --resolve en.wikipedia.org:443:$(kubectl get service my-wikipedia -o jsonpath='{.spec.clusterIP}') https://en.wikipedia.org/wiki/Main_Page | grep -o "<title>.*</title>"
+    <title>Wikipedia, the free encyclopedia</title>
+    {{< /text >}}
+
+1.  In this case, the workload send HTTPS requests (open TLS connection) to the `wikipedia.org`. The traffic is already
+    encrypted by the workload so you can safely disable Istio's mutual TLS:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: DestinationRule
+    metadata:
+      name: my-wikipedia
+    spec:
+      host: my-wikipedia.default.svc.cluster.local
+      trafficPolicy:
+        tls:
+          mode: DISABLE
+    EOF
+    {{< /text >}}
+
+1.  Access `wikipedia.org` by your Kubernetes service's cluster IP from the source pod with Istio sidecar:
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl -s --resolve en.wikipedia.org:443:$(kubectl get service my-wikipedia -o jsonpath='{.spec.clusterIP}') https://en.wikipedia.org/wiki/Main_Page | grep -o "<title>.*</title>"
+    <title>Wikipedia, the free encyclopedia</title>
+    {{< /text >}}
+
+1.  Check that the access is indeed performed by the cluster IP. Notice the sentence
+    `Connected to en.wikipedia.org   (172.21.156.230)` in the output of `curl -v`, it mentions the IP that was printed
+    in the output of your service as the cluster IP.
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl -v --resolve en.wikipedia.org:443:$(kubectl get service my-wikipedia -o jsonpath='{.spec.clusterIP}') https://en.wikipedia.org/wiki/Main_Page -o /dev/null
+    * Added en.wikipedia.org:443:172.21.156.230 to DNS cache
+    * Hostname en.wikipedia.org was found in DNS cache
+    *   Trying 172.21.156.230...
+    * TCP_NODELAY set
+    * Connected to en.wikipedia.org (172.21.156.230) port 443 (#0)
+    ...
+    {{< /text >}}
```

Comment:
a little lost here, if the app can already do TLS, what is the value of using sidecar here?

Reply:
The value of using sidecar: telemetry and policies, based on SNI.